src/redux/books/books.js

```diff
@@ -0,0 +1,25 @@
+function removeBook(state, id) {
+  if (id === state.length - 1) return state.slice(0, id);
+
+  if (id === 0) return state.slice(1);
+
+  return [...state.slice(0, id), ...state.slice(id + 1)];
+}
+
+function addBook(state, book) {
+  return [...state, book];
+}
+
+export default function bookReducer(state = [], action) {
+  switch (action.type) {
+    case 'ADD_BOOK':
+      return addBook(state, action.book);
+    case 'REMOVE_BOOK':
+      return removeBook(state, action.id);
+    default:
+      return state;
+  }
```

Comment:
Nice implementation of the action types 'ADD_BOOK' and 'REMOVE_BOOK'. However, you need to define them before using. The value of each action type needs to follow the [ducks pattern](https://github.com/erikras/ducks-modular-redux), so it will be something like: 
`const ADD_BOOK = bookstore/books/ADD_BOOK` and `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK`

Reply:
Thank you very much, this was very helpful